pkg/aggregate/aggregate.go

```diff
@@ -0,0 +1,73 @@
+package aggregate
+
+import (
+	"github.com/pkg/errors"
+
+	dto "github.com/prometheus/client_model/go"
+
+	"github.com/digitalocean/do-agent/pkg/clients/tsclient"
+)
+
+// MetricWithValue is a representation of a label formatted metric with a value
+type MetricWithValue struct {
+	LFM   map[string]string
+	Value float64
+}
+
+// Aggregate aggregates metric families according to the given aggregate spec.
+// A spec with key: {"metricName": "aggregateLabel"} will remove the "aggregateLabel" from all
+// "metricName" metric families
+func Aggregate(metrics []*dto.MetricFamily, aggregateSpec map[string]string) ([]MetricWithValue, error) {
+	agg := map[string]MetricWithValue{}
+	for _, mf := range metrics {
+		for _, metric := range mf.Metric {
+			var value float64
+			switch *mf.Type {
+			case dto.MetricType_GAUGE:
+				value = *metric.Gauge.Value
+			case dto.MetricType_COUNTER:
+				value = *metric.Counter.Value
+			case dto.MetricType_UNTYPED:
+				value = *metric.Untyped.Value
+			default:
+				// FIXME -- expand this to support other types
```

Comment:
FIXME ? Do we need this fixed now?

Reply:
this was copied over from sonar.go, so it was someone else's. We actually don't need anything there as we don't support historgrams or anything besides the above three types.

Reply:
This is sprinkled throughout our codebase where we do this switch/case and we have never contemplated adding more types so I think we can kill this FIXME

Reply:
perfect, I'm gonna axe it then